### ISSUE TITLE
question on loading an intermediate snapshot of a model and steps_per_epoch

### ISSUE THREAD

**Issue by thadpasce16** [ppwwyyxx/tensorpack#79]
Question regarding what to pass to --load.

When I run a training process, say imagenet-resnet, I get a model snapshot every epoch.

I see something like this:
model-10000.index
model-10000.data-00000-of-00001
model-15000.index
model-15000.data-00000-of-00001
...and so on.

Say I want to restart my training process from 25000 steps, what exactly should I pass as argument to the --load parameter?
Is this correct?
`python imagenet-resnet.py --gpu 0,1 --data /ssd/imagenet -d 18 --load model-25000.index`

What are the .index and .data files? The data files say XYZ.data-00000-of-00001, I dont see a XYZ.data-00001-of-00001? Am I missing something, or one file didnt get saved to my disk during the training process?

In imagenet the training set has 1,281,167 image files. With total batch size of 256 and 5000 steps per epoch, we cover 1,280,000 every epoch. I.e. we dont consider ~1100 images in an epoch. Should this be correct?

**Comment by ppwwyyxx** [comment]
These are tensorflow Saver V2 format. You can load it by --load path/to/model-25000. See https://github.com/tensorflow/tensorflow/issues/6142 for more information (and some caveats).

5000 is just a random number I chose. All images will be trained because:
1. ImageNet data get shuffled every time it is exhausted.
2. Even if it didn't do the shuffle, the data didn't get reset after every epoch, i.e. in the next epoch, the first batch will start from 128000th image.

**Comment by thadpasce16** [comment]
Thanks.

Is there a way to specify the logdir to store the results/log files/snapshots etc rather than have the script pick up the log dir name itself?

If the user doesnt specify a logdir then default to the behavior currently implemented. If user specifies a logdir then use that.

**Comment by ppwwyyxx** [comment]
Of course you can. Currently the default directory is set by `logger.auto_set_dir`. You can delete that line and use `logger.set_logger_dir('some/directory')`.

**Comment by thadpasce16** [comment]
Thanks.
